From 84a15f24e558b14dba3f0be939dee682c1de3eea Mon Sep 17 00:00:00 2001
From: cheney <11722200@qq.com>
Date: Fri, 16 May 2025 13:58:38 +0800
Subject: [PATCH 291/533] Changelogs: 1. Enable LPCPU debug print on die1. 2.
 Do not mask this GPIO when MCPU enters low-power mode.

---
 .../boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts  |  6 +++++-
 drivers/soc/eswin/eswin-lpcpu.c                        | 10 ++++++++++
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
index b2fc8303abda..c5153c0d6b9f 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
@@ -488,7 +488,11 @@ &d1_ipc_scpu {
 
 &d0_lpcpu {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_die0_gpio0_default>;
+	irq-gpios = <&porta 0 GPIO_ACTIVE_HIGH>;
 };
+
 &d1_lpcpu {
 	status = "disabled";
 };
@@ -1011,7 +1015,7 @@ &d0_uart0 {
 };
 
 &d1_uart0 {
-	status = "okay";
+	status = "disabled";
 };
 
 &d0_uart1 {
diff --git a/drivers/soc/eswin/eswin-lpcpu.c b/drivers/soc/eswin/eswin-lpcpu.c
index 905410ed89ab..62b6feba70f2 100644
--- a/drivers/soc/eswin/eswin-lpcpu.c
+++ b/drivers/soc/eswin/eswin-lpcpu.c
@@ -41,6 +41,8 @@
 // #include <linux/mailbox/eswin-ipc-scpu.h>
 #include <linux/eswin-win2030-sid-cfg.h>
 #include <soc/eswin/eswin-lpcpu.h>
+#include <linux/gpio.h>
+#include <linux/gpio/consumer.h>
 
 #define LPCPU_FW_RESERVED
 #define FW_BOOT_ADDR 0x80000000
@@ -73,6 +75,7 @@ struct lpcpu_dev {
 	struct reset_control *dbg_rst;
 	void __iomem *mmio;
 	size_t fw_size;
+	struct gpio_desc *irq_gpio;
 };
 
 static struct lpcpu_dev *lpcpu;
@@ -316,6 +319,13 @@ static int eswin_lpcpu_probe(struct platform_device *pdev)
 		return -ENOMEM;
 	platform_set_drvdata(pdev, lpcpu);
 
+	lpcpu->irq_gpio = devm_gpiod_get(dev, "irq", GPIOD_IN);
+	if (IS_ERR(lpcpu->irq_gpio)) {
+		dev_err(dev, "Failed to get IRQ GPIO\n");
+		return PTR_ERR(lpcpu->irq_gpio);
+	}
+	enable_irq_wake(gpiod_to_irq(lpcpu->irq_gpio));
+
 	lpcpu->mdev.minor = MISC_DYNAMIC_MINOR;
 	lpcpu->mdev.name = "lpcpu";
 	lpcpu->mdev.fops = &eswin_lpcpu_ops;
-- 
2.51.1

